pkg/controller/sub_controller/sub_controller.go

```diff
@@ -56,21 +56,90 @@ type SubDefaultController struct {
 	K8srecorder record.EventRecorder
 }
 
+func (d *SubDefaultController) CheckRestartTimeAndInject(dcr *dorisv1.DorisCluster, componentType dorisv1.ComponentType) bool {
+	var baseSpec *dorisv1.BaseSpec
+	var restartedAt string
+	var restartAnnotationsKey string
+	switch componentType {
+	case dorisv1.Component_FE:
+		baseSpec = &dcr.Spec.FeSpec.BaseSpec
+		restartedAt = dcr.Annotations[dorisv1.FERestartAt]
+		restartAnnotationsKey = dorisv1.FERestartAt
+	case dorisv1.Component_BE:
+		baseSpec = &dcr.Spec.BeSpec.BaseSpec
+		restartedAt = dcr.Annotations[dorisv1.BERestartAt]
+		restartAnnotationsKey = dorisv1.BERestartAt
+	default:
+		klog.Errorf("CheckRestartTimeAndInject dorisClusterName %s, namespace %s componentType %s not supported.", dcr.Name, dcr.Namespace, componentType)
+	}
+
+	if restartedAt == "" {
+		return false
+	}
+
+	// run shell: date +"%Y-%m-%dT%H:%M:%S%:z"
+	// "2024-11-21T11:08:56+08:00"
+	parseTime, err := time.Parse(time.RFC3339, restartedAt)
+	if err != nil {
+		checkErr := fmt.Errorf("CheckRestartTimeAndInject error: time format is incorrect. dorisClusterName: %s, namespace: %s, componentType %s, wrong parse 'restartedAt': %s , error: %s", dcr.Name, dcr.Namespace, componentType, restartedAt, err.Error())
+		klog.Error(checkErr.Error())
+		d.K8srecorder.Event(dcr, string(EventWarning), string(RestartParameterIllegal), checkErr.Error())
+		return false
+	}
+
+	restartAt := time.Now().Add(-10 * time.Minute)
+
+	if restartAt.After(parseTime) {
+		checkErr := fmt.Errorf("CheckRestartTimeAndInject The time has expired, dorisClusterName: %s, namespace: %s, componentType %s, wrong parse 'restartedAt': %s : The time has expired, if you want to restart doris, please set a future time", dcr.Name, dcr.Namespace, componentType, restartedAt)
+		klog.Error(checkErr.Error())
```

Comment:
the event message is too long. 
"the fe or be restart time is not effective.  the restart time can't be earlier than 10 minutes before the current time."